Automattic/newspack-blocks
# feat(donate): agree to pay fees feature

### All Submissions:






### Changes proposed in this Pull Request:



Adds a "Agree to pay fees" option to Donate block. Available only in the streamlined (Stripe) version.

<img width="567" alt="image" src="https://user-images.githubusercontent.com/7383192/142845112-77bcaf04-1250-4492-aa25-ed85e991138b.png">

If user has a non-standard Stripe fee (e.g. [is a non-profit](thttps://support.stripe.com/questions/fee-discount-for-nonprofit-organizations)), they can update it via options. UI for this is coming soon, for now these should be updated manually (`wp option set …`).

### How to test the changes in this Pull Request:

1. Enable AMP Plus and configure platform as Stripe in Reader Revenue wizard
2. Visit a page with the Donate block
3. Observe a new input in the form – "Agree to pay fees"
4. The input should be checked on by default and inform about the standard Stripe fee, which is 2.9% + ¢30
5. Test the donation with and without the input checked
6. Update the fee variables by setting `newspack_blocks_donate_fee_multiplier` & `newspack_blocks_donate_fee_static` options 

### Other information:
